client/asset/firo/firo.go

```diff
@@ -154,18 +165,29 @@ func NewWallet(cfg *asset.WalletConfig, logger dex.Logger, network dex.Network)
 		NumericGetRawRPC:         false, // getrawtransaction uses either 0/1 Or true/false
 		LegacyValidateAddressRPC: true,  // use validateaddress to read 'ismine' bool
 		SingularWallet:           true,  // one wallet/node
-		UnlockSpends:             false, // checked after sendtoaddress
+		UnlockSpends:             true,  // does Not unlock coins after sendrawtransaction
 		AssetID:                  BipID,
 		FeeEstimator:             estimateFee,
 		ExternalFeeEstimator:     fetchExternalFee,
 		PrivKeyFunc: func(addr string) (*btcec.PrivateKey, error) {
-			return privKeyForAddress(w, addr)
+			return privKeyForAddress(exw, addr)
 		},
 	}
 
-	var err error
-	w, err = btc.BTCCloneWallet(cloneCFG)
-	return w, err
+	switch cfg.Type {
+	case walletTypeRPC, walletTypeLegacy:
+		var err error
+		exw, err = btc.BTCCloneWallet(cloneCFG)
+		return exw, err
```

Comment:
Can just `return btc.BTCCloneWallet(cloneCFG)`?

Reply:
Can return, but later we will panic because we are not an `ExchangeWalletFullNode` when we callback `privKeyForAddress(rpcCaller, ...` here in the same file. 


<details>
<summary>Changed code</summary>

![image](https://github.com/decred/dcrdex/assets/126670673/484716e0-3b81-47de-8832-edc37bcbcadd)
</details>

<details>
<summary>Stacktrace</summary>

```
 7  0x0000000000a1ffb1 in decred.org/dcrdex/client/asset/btc.(*ExchangeWalletFullNode).CallRPC
     at <autogenerated>:1
 8  0x0000000000a31db4 in decred.org/dcrdex/client/asset/firo.privKeyForAddress
     at /home/dev/dex/dcrdex/client/asset/firo/firo.go:206
 9  0x0000000000a31bb0 in decred.org/dcrdex/client/asset/firo.NewWallet.func1
     at /home/dev/dex/dcrdex/client/asset/firo/firo.go:173
10  0x0000000000a14e14 in decred.org/dcrdex/client/asset/btc.(*rpcClient).privKeyForAddress
     at /home/dev/dex/dcrdex/client/asset/btc/rpcclient.go:649

```
</details>